dtmsvr/storage/sql/sql.go

```diff
@@ -164,11 +170,16 @@ func (s *Store) LockOneGlobalTrans(expireIn time.Duration) *storage.TransGlobalS
 	where := fmt.Sprintf(`next_cron_time < '%s' and status in ('prepared', 'aborting', 'submitted')`, nextCronTime)
 
 	order := map[string]string{
-		dtmimp.DBTypeMysql:    `order by rand()`,
-		dtmimp.DBTypePostgres: `order by random()`,
+		dtmimp.DBTypeMysql:     `order by rand()`,
+		dtmimp.DBTypePostgres:  `order by random()`,
+		dtmimp.DBTypeSqlServer: `order by rand()`,
 	}[conf.Store.Driver]
 
-	ssql := fmt.Sprintf(`select id from trans_global where %s %s limit 1`, where, order)
+	ssql := map[string]string{
+		dtmimp.DBTypeMysql:     fmt.Sprintf(`select id from trans_global where %s %s limit 1`, where, order),
+		dtmimp.DBTypePostgres:  fmt.Sprintf(`select id from trans_global where %s %s limit 1`, where, order),
+		dtmimp.DBTypeSqlServer: fmt.Sprintf(`select top 1 id from trans_global where %s %s`, where, order),
+	}[conf.Store.Driver]
```

Comment:
Since we added sqlserver, then we can eliminate variable `order`, just use one variable `ssql` and add order part in it.